angular/src/app/modules/timesheet/timesheet-detail/timesheet-detail.component.ts

```diff
@@ -273,24 +273,31 @@ export class TimesheetDetailComponent extends PagedListingComponentBase<Timeshee
   //     })
   // }
 
-  exportTimesheetDetail(exportInvoiceMode) {
-    let payloadDto = {
-      timesheetId: this.timesheetId,
-      projectIds: this.listExportInvoice,
-      mode: exportInvoiceMode
-    }
+  exportTimesheetDetail(item: any) {
+    let payloadDto;
+    if (item && item.projectId) {
+      payloadDto = {
+        timesheetId: this.timesheetId,
+        projectIds: [item.projectId],
+        mode: this.APP_ENUM.ExportInvoiceMode.Normal
+      };
+  } else {
+      payloadDto = {
+        timesheetId: this.timesheetId,
+        projectIds: this.listExportInvoice,
+        mode: this.APP_ENUM.ExportInvoiceMode.Normal
+    };
+  }
     this.timesheetProjectService.exportTimeSheetDetail(payloadDto).subscribe((res) => {
-      const file = new Blob([this.s2ab(atob(res.result.base64))], {
-        type: "application/vnd.ms-excel;charset=utf-8"
-      });
+    const file = new Blob([this.s2ab(atob(res.result.base64))], {
+      type: "application/vnd.ms-excel;charset=utf-8"
+    });
       this.refresh();
-      this.listExportInvoice=[];
       FileSaver.saveAs(file, res.result.fileName);
       abp.notify.success("Export Invoice For Tax Successfully!");
```

Comment:
abp.notify.success("Export Invoice For Tax Successfully!");
=>
abp.notify.success("Export TS Detail Successfully!");